### PR TITLE
Propagate textfield character limits to semantics

### LINKED ISSUE #38940: a11y: TextField with maxLength has a11y problems
Internal: b/140829380

If you have a text field with maxLength, such as

`TextField(maxLength: 3, decoration: InputDecoration(hintText: "hint text test"))`

it behaves poorly in TalkBack.

When I try to navigate to a TextField like this, it focuses only on the 0/3 character count, skipping the input field. The character count reads nicely, announcing "3 characters remaining", but the hintText is not read and it does not indicate that a text field is in focus and can be interacted with. If you double-tap now, it will open the keyboard, but the TalkBack outline is still around the 0/3. I guess this is in some kind of bad initial state.

Now, type some text. When you type the 4th character and exceed the limit, it correctly does not add the character, but it just reads "Showing English (US)(QUERTY) keyboard". I would expect it to provide some indication that the 4th character has been rejected, like "no characters remaining".

Now navigate backwards and then forwards again. Now that the text field has text in it, it reads "Editing, xxx, hint text test, edit box". You can then navigate forwards again to focus on the 3/3, which will properly read "No characters remaining".

If you delete all of the input text, it does not go back into the bad initial state where the screen reader does not see the input field.

### PR DESCRIPTION
**Pull request by goderbauer**
## Description

What the title says. Corresponding engine change: https://github.com/flutter/engine/pull/12269.

Also fixes an issue where you could interact with a disabled textfield via talkback.

## Related Issues

Fixes flutter/flutter#38940.

## Tests

I added the following tests:

* Cannot interact with disabled text field
* character limits are propagated properly. 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 1]
```diff
@@ -309,6 +311,20 @@ class SemanticsData extends Diagnosticable {
   ///  * [UiKitView], which is the platform view for iOS.
   final int platformViewId;
 
+  /// The maximum number of characters that can be entered into an editable
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/semantics/semantics.dart]: We should specify exactly what we mean by "characters" here

**Review by jonahwilliams — APPROVED** [review]
LGTM with nit.

We should be very specific about how the framework counts characters here.

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/src/semantics/semantics.dart]: Added clarification.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**